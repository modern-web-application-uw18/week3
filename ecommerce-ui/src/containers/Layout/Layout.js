import React, { Component } from 'react';
import Header from '../../components/Header/Header'
import classes from './Layout.css';
import mockDataArr from '../../assets/airbnbs.json';
import Card from '../../components/Card/Card';

class Layout extends Component {

    state = {
        mockDataArr: mockDataArr,
        cartContent: []
    }


    addListingToTheCartHandler = (listing) => {
        this.setState((prevState, props) => {
            let transformedState = [...prevState.cartContent];
            transformedState.push(listing);
            return { cartContent: transformedState }
        });
    }

    removeListingFromTheCartHandler = (listing) => {
        this.setState((prevState, props) => {
            let transformedState = [...prevState.cartContent];

            transformedState.filter(
                (obj, idx) => obj.title === listing.title ? transformedState.splice(idx, 1) : null //remove item from the array based on the title
            );
            return { cartContent: transformedState }

        });
    }


    mockDataHandler = (data) => {
        this.setState((prevState, props) => {
            let newState = [...prevState.mockDataArr];

            //setting default data in case of any of the elements missing
            let transformedData = {
                title: data.title ? data.title : 'New Listing',
                houseType: data.houseType ? data.houseType : 'Apartment',
                image: data.image ? data.image : 'https://cdn.geekwire.com/wp-content/uploads/2017/03/seattle-630x359.jpg',
                location: {
                    city: data.city ? data.city : "City",
                    country: data.country ? data.countr : "Country"
                },
                payment: {
                    cost: data.cost ? parseInt(data.cost,10) : 0, //making sure this is an integer to avoid issues when calculating total
                },
                host: {
                    name: data.name ? data.name : 'Host',
                },
                rating: {
                    stars: 0,
                    reviews: 0
                }
            }

            newState.push(transformedData);
            return { mockDataArr: newState };
        });

    }


    render() {

        return (
            <div className={classes.Layout}>

                <Header
                    dataHandler={this.mockDataHandler}
                    cartContent={this.state.cartContent}
                    removeListingFromTheCartHandler={this.removeListingFromTheCartHandler}
                    mockDataArr={this.state.mockDataArr}
                />

                <main className={classes.MainStage}>


                    {this.state.mockDataArr.map((item, idx) => {
                        //only show listings that are not in the cart already
                        return !this.state.cartContent.filter((obj) => obj.title === item.title).length ? <Card
                            item={item}
                            key={idx}
                            addListingToTheCartHandler={this.addListingToTheCartHandler}
                        /> : null;
                    })}



                </main>

            </div>
        )
    }
}

export default Layout;
