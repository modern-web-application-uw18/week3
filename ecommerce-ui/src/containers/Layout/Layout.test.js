import React from 'react';
import { shallow, mount} from 'enzyme';
import Layout from './Layout';
import Header from '../../components/Header/Header'
import classes from './Layout.css';
import mockDataArr from '../../assets/airbnbs.json';
import Card from '../../components/Card/Card';

describe('<Layout />', () => {

    let mockListing = {
        "title": "Centrally-located Manhattan studio",
        "houseType": "Apartment",
        "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
        "location": {
            "city": "New York",
            "country": "USA"
        },
        "payment": {
            "cost": 105,
            "description": "Free cancellation"
        },
        "host": {
            "name": "Robert",
            "isSuperhost": true
        },
        "rating": {
            "stars": 3,
            "reviews": 133
        }
    }

    it('Snapshot test', () => {
        const wrapper = mount(<Layout  />);

        wrapper.instance().mockDataHandler(mockListing);
        wrapper.instance().removeListingFromTheCartHandler(mockListing);
        wrapper.instance().addListingToTheCartHandler(mockListing);

        expect(wrapper).toMatchSnapshot();
    })
})