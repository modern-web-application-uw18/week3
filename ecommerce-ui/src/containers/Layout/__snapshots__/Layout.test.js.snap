// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> Snapshot test 1`] = `
<Layout>
  <div>
    <Header
      cartContent={Array []}
      dataHandler={[Function]}
      mockDataArr={
        Array [
          Object {
            "host": Object {
              "isSuperhost": true,
              "name": "Robert",
            },
            "houseType": "Apartment",
            "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "New York",
              "country": "USA",
            },
            "payment": Object {
              "cost": 105,
              "description": "Free cancellation",
            },
            "rating": Object {
              "reviews": 133,
              "stars": 3,
            },
            "title": "Centrally-located Manhattan studio",
          },
          Object {
            "host": Object {
              "isSuperhost": true,
              "name": "Jamie",
            },
            "houseType": "Apartment",
            "image": "https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "Denver",
              "country": "USA",
            },
            "payment": Object {
              "cost": 130,
              "description": "",
            },
            "rating": Object {
              "reviews": 381,
              "stars": 4.5,
            },
            "title": "Denver apartment outside Union Station",
          },
          Object {
            "host": Object {
              "isSuperhost": false,
              "name": "Olivia",
            },
            "houseType": "Entire House",
            "image": "https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "Stockholm",
              "country": "Sweden",
            },
            "payment": Object {
              "cost": 230,
              "description": "Free cancellation",
            },
            "rating": Object {
              "reviews": 83,
              "stars": 5,
            },
            "title": "Quaint 3-bedroom home in Gamla stan",
          },
        ]
      }
      removeListingFromTheCartHandler={[Function]}
    >
      <div>
        <div>
          <button
            className="btn btn-danger"
            onClick={[Function]}
            style={
              Object {
                "border": "0px",
                "height": "40px",
                "width": "150px",
              }
            }
            type="button"
          >
            Add Listing    
            <MdAddBox
              size={30}
            >
              <IconBase
                size={30}
                viewBox="0 0 40 40"
              >
                <svg
                  fill="currentColor"
                  height={30}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": undefined,
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={30}
                >
                  <g>
                    <path
                      d="m28.4 21.6v-3.2h-6.8v-6.8h-3.2v6.8h-6.8v3.2h6.8v6.8h3.2v-6.8h6.8z m3.2-16.6c1.8 0 3.4 1.6 3.4 3.4v23.2c0 1.8-1.6 3.4-3.4 3.4h-23.2c-1.9 0-3.4-1.6-3.4-3.4v-23.2c0-1.8 1.5-3.4 3.4-3.4h23.2z"
                    />
                  </g>
                </svg>
              </IconBase>
            </MdAddBox>
          </button>
        </div>
        <div
          style={
            Object {
              "textAlign": "right",
            }
          }
        >
          <button
            className="btn btn-default"
            onClick={[Function]}
            style={
              Object {
                "border": "0px",
                "height": "40px",
                "width": "150px",
              }
            }
            type="button"
          >
            Checkout    
            <MdShoppingCart
              size={30}
            >
              <IconBase
                size={30}
                viewBox="0 0 40 40"
              >
                <svg
                  fill="currentColor"
                  height={30}
                  preserveAspectRatio="xMidYMid meet"
                  style={
                    Object {
                      "color": undefined,
                      "verticalAlign": "middle",
                    }
                  }
                  viewBox="0 0 40 40"
                  width={30}
                >
                  <g>
                    <path
                      d="m28.4 30c1.8 0 3.2 1.6 3.2 3.4s-1.4 3.2-3.2 3.2-3.4-1.4-3.4-3.2 1.6-3.4 3.4-3.4z m-26.8-26.6h5.5l1.6 3.2h24.7c0.9 0 1.6 0.8 1.6 1.8 0 0.3-0.1 0.5-0.2 0.7l-6 10.8c-0.5 1-1.6 1.7-2.9 1.7h-12.4l-1.5 2.8 0 0.2c0 0.2 0.1 0.4 0.3 0.4h19.3v3.4h-20c-1.8 0-3.2-1.6-3.2-3.4 0-0.5 0.1-1.1 0.3-1.6l2.3-4.1-6-12.7h-3.4v-3.2z m10 26.6c1.8 0 3.4 1.6 3.4 3.4s-1.6 3.2-3.4 3.2-3.2-1.4-3.2-3.2 1.4-3.4 3.2-3.4z"
                    />
                  </g>
                </svg>
              </IconBase>
            </MdShoppingCart>
          </button>
        </div>
        <cart
          cartContent={Array []}
          cartShow={false}
          handleCartClose={[Function]}
          removeListingFromTheCartHandler={[Function]}
        >
          <Modal
            animation={true}
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <Modal
              autoFocus={true}
              backdrop={true}
              backdropClassName="modal-backdrop"
              backdropTransition={[Function]}
              containerClassName="modal-open"
              enforceFocus={true}
              keyboard={true}
              manager={
                ModalManager {
                  "add": [Function],
                  "containers": Array [],
                  "data": Array [],
                  "handleContainerOverflow": true,
                  "hideSiblingNodes": true,
                  "isTopModal": [Function],
                  "modals": Array [],
                  "remove": [Function],
                }
              }
              onEntering={[Function]}
              onExited={[Function]}
              onHide={[Function]}
              renderBackdrop={[Function]}
              restoreFocus={true}
              show={false}
              transition={[Function]}
            />
          </Modal>
        </cart>
        <NewListingModal
          dataHandler={[Function]}
          formShow={false}
          handleFormClose={[Function]}
          mockDataArr={
            Array [
              Object {
                "host": Object {
                  "isSuperhost": true,
                  "name": "Robert",
                },
                "houseType": "Apartment",
                "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
                "location": Object {
                  "city": "New York",
                  "country": "USA",
                },
                "payment": Object {
                  "cost": 105,
                  "description": "Free cancellation",
                },
                "rating": Object {
                  "reviews": 133,
                  "stars": 3,
                },
                "title": "Centrally-located Manhattan studio",
              },
              Object {
                "host": Object {
                  "isSuperhost": true,
                  "name": "Jamie",
                },
                "houseType": "Apartment",
                "image": "https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80",
                "location": Object {
                  "city": "Denver",
                  "country": "USA",
                },
                "payment": Object {
                  "cost": 130,
                  "description": "",
                },
                "rating": Object {
                  "reviews": 381,
                  "stars": 4.5,
                },
                "title": "Denver apartment outside Union Station",
              },
              Object {
                "host": Object {
                  "isSuperhost": false,
                  "name": "Olivia",
                },
                "houseType": "Entire House",
                "image": "https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80",
                "location": Object {
                  "city": "Stockholm",
                  "country": "Sweden",
                },
                "payment": Object {
                  "cost": 230,
                  "description": "Free cancellation",
                },
                "rating": Object {
                  "reviews": 83,
                  "stars": 5,
                },
                "title": "Quaint 3-bedroom home in Gamla stan",
              },
            ]
          }
        >
          <Modal
            animation={true}
            autoFocus={true}
            backdrop={true}
            bsClass="modal"
            dialogComponentClass={[Function]}
            enforceFocus={true}
            keyboard={true}
            manager={
              ModalManager {
                "add": [Function],
                "containers": Array [],
                "data": Array [],
                "handleContainerOverflow": true,
                "hideSiblingNodes": true,
                "isTopModal": [Function],
                "modals": Array [],
                "remove": [Function],
              }
            }
            onHide={[Function]}
            renderBackdrop={[Function]}
            restoreFocus={true}
            show={false}
          >
            <Modal
              autoFocus={true}
              backdrop={true}
              backdropClassName="modal-backdrop"
              backdropTransition={[Function]}
              containerClassName="modal-open"
              enforceFocus={true}
              keyboard={true}
              manager={
                ModalManager {
                  "add": [Function],
                  "containers": Array [],
                  "data": Array [],
                  "handleContainerOverflow": true,
                  "hideSiblingNodes": true,
                  "isTopModal": [Function],
                  "modals": Array [],
                  "remove": [Function],
                }
              }
              onEntering={[Function]}
              onExited={[Function]}
              onHide={[Function]}
              renderBackdrop={[Function]}
              restoreFocus={true}
              show={false}
              transition={[Function]}
            />
          </Modal>
        </NewListingModal>
      </div>
    </Header>
    <main>
      <card
        addListingToTheCartHandler={[Function]}
        item={
          Object {
            "host": Object {
              "isSuperhost": true,
              "name": "Robert",
            },
            "houseType": "Apartment",
            "image": "https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "New York",
              "country": "USA",
            },
            "payment": Object {
              "cost": 105,
              "description": "Free cancellation",
            },
            "rating": Object {
              "reviews": 133,
              "stars": 3,
            },
            "title": "Centrally-located Manhattan studio",
          }
        }
        key="0"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(https://images.unsplash.com/photo-1490806230066-f7e6f7bf5052?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=2ca889535ea01f912f94ac4ddf0034e0&auto=format&fit=crop&w=500&q=80)",
            }
          }
        >
          <div>
            Centrally-located Manhattan studio
          </div>
          <div>
            <div>
              $ 
              105
            </div>
          </div>
          <div>
            <div
              style={
                Object {
                  "width": "200px",
                }
              }
            >
              Apartment
            </div>
              
            <star
              starCount={3}
            >
              <div>
                <MdStar
                  key="0"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="1"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="2"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
              </div>
            </star>
            <span>
              (
              133
              )
            </span>
          </div>
          <div>
            <div>
              New York
              , 
              USA
            </div>
            <div
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <div>
                Robert
              </div>
            </div>
            <div
              style={
                Object {
                  "textAlign": "right",
                }
              }
            >
              <button
                className="btn btn-default"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#5b5b5b",
                    "border": "0px",
                    "height": "35px",
                    "width": "45px",
                  }
                }
                type="button"
              >
                <MdAddShoppingCart
                  size={20}
                  style={
                    Object {
                      "color": "#ffffff",
                    }
                  }
                >
                  <IconBase
                    size={20}
                    style={
                      Object {
                        "color": "#ffffff",
                      }
                    }
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height={20}
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": "#ffffff",
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width={20}
                    >
                      <g>
                        <path
                          d="m12 24.6q0 0.4 0.3 0.4h19.3v3.4h-20q-1.3 0-2.3-1.1t-0.9-2.3q0-0.8 0.4-1.6l2.2-4.1-6-12.7h-3.4v-3.2h5.5q1.6 3.2 3.1 6.6 0.4 0.7 1.8 3.7t2.2 4.7h11.7q5.9-10.7 6.4-11.8l2.9 1.7-6.4 11.6q-0.9 1.7-2.9 1.7h-12.4l-1.5 2.8z m16.4 5.4q1.3 0 2.3 1t0.9 2.4-0.9 2.3-2.3 0.9-2.4-0.9-1-2.3 1-2.4 2.4-1z m-16.8 0q1.4 0 2.4 1t1 2.4-1 2.3-2.4 0.9-2.3-0.9-0.9-2.3 0.9-2.4 2.3-1z m6.8-15v-5h-5v-3.4h5v-5h3.2v5h5v3.4h-5v5h-3.2z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdAddShoppingCart>
              </button>
            </div>
          </div>
        </div>
      </card>
      <card
        addListingToTheCartHandler={[Function]}
        item={
          Object {
            "host": Object {
              "isSuperhost": true,
              "name": "Jamie",
            },
            "houseType": "Apartment",
            "image": "https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "Denver",
              "country": "USA",
            },
            "payment": Object {
              "cost": 130,
              "description": "",
            },
            "rating": Object {
              "reviews": 381,
              "stars": 4.5,
            },
            "title": "Denver apartment outside Union Station",
          }
        }
        key="1"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(https://images.unsplash.com/photo-1519424187720-db6d0fc5a5d2?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=19f783ae564d7c6da4777cde1a84c1c9&auto=format&fit=crop&w=500&q=80)",
            }
          }
        >
          <div>
            Denver apartment outside Union Station
          </div>
          <div>
            <div>
              $ 
              130
            </div>
          </div>
          <div>
            <div
              style={
                Object {
                  "width": "200px",
                }
              }
            >
              Apartment
            </div>
              
            <star
              starCount={4.5}
            >
              <div>
                <MdStar
                  key="0"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="1"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="2"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="3"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStarHalf
                  key="4"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 25.7l6.3 3.8-1.7-7.2 5.6-4.8-7.4-0.6-2.8-6.7v15.5z m16.6-10.3l-9 7.9 2.7 11.7-10.3-6.2-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStarHalf>
              </div>
            </star>
            <span>
              (
              381
              )
            </span>
          </div>
          <div>
            <div>
              Denver
              , 
              USA
            </div>
            <div
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <div>
                Jamie
              </div>
            </div>
            <div
              style={
                Object {
                  "textAlign": "right",
                }
              }
            >
              <button
                className="btn btn-default"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#5b5b5b",
                    "border": "0px",
                    "height": "35px",
                    "width": "45px",
                  }
                }
                type="button"
              >
                <MdAddShoppingCart
                  size={20}
                  style={
                    Object {
                      "color": "#ffffff",
                    }
                  }
                >
                  <IconBase
                    size={20}
                    style={
                      Object {
                        "color": "#ffffff",
                      }
                    }
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height={20}
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": "#ffffff",
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width={20}
                    >
                      <g>
                        <path
                          d="m12 24.6q0 0.4 0.3 0.4h19.3v3.4h-20q-1.3 0-2.3-1.1t-0.9-2.3q0-0.8 0.4-1.6l2.2-4.1-6-12.7h-3.4v-3.2h5.5q1.6 3.2 3.1 6.6 0.4 0.7 1.8 3.7t2.2 4.7h11.7q5.9-10.7 6.4-11.8l2.9 1.7-6.4 11.6q-0.9 1.7-2.9 1.7h-12.4l-1.5 2.8z m16.4 5.4q1.3 0 2.3 1t0.9 2.4-0.9 2.3-2.3 0.9-2.4-0.9-1-2.3 1-2.4 2.4-1z m-16.8 0q1.4 0 2.4 1t1 2.4-1 2.3-2.4 0.9-2.3-0.9-0.9-2.3 0.9-2.4 2.3-1z m6.8-15v-5h-5v-3.4h5v-5h3.2v5h5v3.4h-5v5h-3.2z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdAddShoppingCart>
              </button>
            </div>
          </div>
        </div>
      </card>
      <card
        addListingToTheCartHandler={[Function]}
        item={
          Object {
            "host": Object {
              "isSuperhost": false,
              "name": "Olivia",
            },
            "houseType": "Entire House",
            "image": "https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80",
            "location": Object {
              "city": "Stockholm",
              "country": "Sweden",
            },
            "payment": Object {
              "cost": 230,
              "description": "Free cancellation",
            },
            "rating": Object {
              "reviews": 83,
              "stars": 5,
            },
            "title": "Quaint 3-bedroom home in Gamla stan",
          }
        }
        key="2"
      >
        <div
          style={
            Object {
              "backgroundImage": "url(https://images.unsplash.com/photo-1495136568018-11a54caad6b5?ixlib=rb-0.3.5&ixid=eyJhcHBfaWQiOjEyMDd9&s=b4d19de84a578b1175e85e0811f41c97&auto=format&fit=crop&w=500&q=80)",
            }
          }
        >
          <div>
            Quaint 3-bedroom home in Gamla stan
          </div>
          <div>
            <div>
              $ 
              230
            </div>
          </div>
          <div>
            <div
              style={
                Object {
                  "width": "200px",
                }
              }
            >
              Entire House
            </div>
              
            <star
              starCount={5}
            >
              <div>
                <MdStar
                  key="0"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="1"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="2"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="3"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
                <MdStar
                  key="4"
                >
                  <IconBase
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height="1em"
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": undefined,
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width="1em"
                    >
                      <g>
                        <path
                          d="m20 28.8l-10.3 6.2 2.7-11.7-9-7.9 11.9-1 4.7-11 4.7 11 11.9 1-9 7.9 2.7 11.7z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdStar>
              </div>
            </star>
            <span>
              (
              83
              )
            </span>
          </div>
          <div>
            <div>
              Stockholm
              , 
              Sweden
            </div>
            <div
              style={
                Object {
                  "textAlign": "left",
                }
              }
            >
              <div>
                Olivia
              </div>
            </div>
            <div
              style={
                Object {
                  "textAlign": "right",
                }
              }
            >
              <button
                className="btn btn-default"
                onClick={[Function]}
                style={
                  Object {
                    "backgroundColor": "#5b5b5b",
                    "border": "0px",
                    "height": "35px",
                    "width": "45px",
                  }
                }
                type="button"
              >
                <MdAddShoppingCart
                  size={20}
                  style={
                    Object {
                      "color": "#ffffff",
                    }
                  }
                >
                  <IconBase
                    size={20}
                    style={
                      Object {
                        "color": "#ffffff",
                      }
                    }
                    viewBox="0 0 40 40"
                  >
                    <svg
                      fill="currentColor"
                      height={20}
                      preserveAspectRatio="xMidYMid meet"
                      style={
                        Object {
                          "color": "#ffffff",
                          "verticalAlign": "middle",
                        }
                      }
                      viewBox="0 0 40 40"
                      width={20}
                    >
                      <g>
                        <path
                          d="m12 24.6q0 0.4 0.3 0.4h19.3v3.4h-20q-1.3 0-2.3-1.1t-0.9-2.3q0-0.8 0.4-1.6l2.2-4.1-6-12.7h-3.4v-3.2h5.5q1.6 3.2 3.1 6.6 0.4 0.7 1.8 3.7t2.2 4.7h11.7q5.9-10.7 6.4-11.8l2.9 1.7-6.4 11.6q-0.9 1.7-2.9 1.7h-12.4l-1.5 2.8z m16.4 5.4q1.3 0 2.3 1t0.9 2.4-0.9 2.3-2.3 0.9-2.4-0.9-1-2.3 1-2.4 2.4-1z m-16.8 0q1.4 0 2.4 1t1 2.4-1 2.3-2.4 0.9-2.3-0.9-0.9-2.3 0.9-2.4 2.3-1z m6.8-15v-5h-5v-3.4h5v-5h3.2v5h5v3.4h-5v5h-3.2z"
                        />
                      </g>
                    </svg>
                  </IconBase>
                </MdAddShoppingCart>
              </button>
            </div>
          </div>
        </div>
      </card>
    </main>
  </div>
</Layout>
`;
